Uplift/Views/Loading/LoadingCollectionView.swift

```diff
@@ -0,0 +1,169 @@
+            layout.sectionInset = UIEdgeInsets(top: 0.0, left: calendarClassSideInsets, bottom: 0.0, right: calendarClassSideInsets)
+            height = (classesCellHeight + classesMinimumLineSpacing) * CGFloat(classesCellCount) - classesMinimumLineSpacing
+            headerWidth = classesHeaderWidth
+        }
+
+        delegate = self
+        dataSource = self
+
+        backgroundColor = .clear
+
+        register(LoadingCollectionViewCell.self, forCellWithReuseIdentifier: Identifiers.loadingCollectionViewCell)
+        register(LoadingCollectionHeaderView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: Identifiers.loadingHeaderView)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+}
+
+extension LoadingCollectionView: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        switch collectionViewType {
+        case .smallGrid:
+            let itemWidth = (collectionViewWidth - leftInset - smallGridMinimumInteritemSpacing - rightInset) / 2.0
+            return CGSize(width: itemWidth, height: smallGridCellHeight)
+        case .horizontallyScrolling:
+            return CGSize(width: horizontallyScrollingCellWidth, height: horizontallyScrollingCellHeight)
+        case .largeGrid:
+            return CGSize(width: largeGridCellWidth, height: largeGridCellHeight)
+        case .calendar:
+            return CGSize(width: calendarCellSize, height: calendarCellSize)
+        case .classes:
+            return CGSize(width: collectionViewWidth - (2 * calendarClassSideInsets), height: classesCellHeight)
+        default:
+            return CGSize(width: 0, height: 0)
+        }
+    }
+
+}
+
+extension LoadingCollectionView: UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        switch collectionViewType {
+        case .smallGrid:
+            return smallGridCellCount
+        case .horizontallyScrolling:
+            return horizontallyScrollingCellCount
+        case .largeGrid:
+            return largeGridCellCount
+        case .calendar:
+            return calendarCellCount
+        case .classes:
+            return classesCellCount
+        case .none:
+            return 0
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = dequeueReusableCell(withReuseIdentifier: Identifiers.loadingCollectionViewCell, for: indexPath) as! LoadingCollectionViewCell
+        if collectionViewType == .calendar {
+            cell.contentView.layer.cornerRadius = calendarCellSize / 2.0
+        }
```

Comment:
should we also set the `cornerRadius` to 0 if the type is not calendar?

Reply:
I think `cornerRadius` is 0 by default

Reply:
it is but since we're reusing cells, we could be using a cell that previously had it set

Reply:
Good point